packages/transformer/src/IModelTransformer.ts

```diff
@@ -1270,12 +1348,14 @@ export class IModelTransformer extends IModelExportHandler {
   protected shouldDetectDeletes(): boolean {
     // FIXME: all synchronizations should mark this as false, but we can probably change this
     // to just follow the new deprecated option
-    if (this._isFirstSynchronization) return false; // not necessary the first time since there are no deletes to detect
+    if (this._isProvenanceInitTransform) return false; // not necessary the first time since there are no deletes to detect
 
-    if (this._options.isReverseSynchronization) return false; // not possible for a reverse synchronization since provenance will be deleted when element is deleted
+    if (this.isReverseSynchronization) return false; // not possible for a reverse synchronization since provenance will be deleted when element is deleted
 
     // FIXME: do any tests fail? if not, consider using @see _isSynchronization
-    if (this._isForwardSynchronization) return false; // not possible for a reverse synchronization since provenance will be deleted when element is deleted
+    // FIXME<NICK> why was this return false before? and why did this work before my changes to determine isReverseSync?
+    // needed to change to true for 'detect element deletes works on children' for this to work.
+    if (this._isForwardSynchronization) return true; // not possible for a reverse synchronization since provenance will be deleted when element is deleted.
```

Comment:
What should the comment be here for forward sync "// not possible for a reverse synchronization since provenance will be deleted when element is deleted", since it appears to be copied from the comment for this.isReverseSync.

Something like "not necessary to brute force detect deletions in a forward sync because we can detect all necessary deletes from changeset data" ?